tests/typecheck/fields/test_custom_fields.yml

```diff
@@ -0,0 +1,62 @@
+-   case: test_custom_model_fields_with_generic_type
+    main: |
+        from myapp.models import User, CustomFieldValue
+        user = User()
+        reveal_type(user.id)  # N: Revealed type is "builtins.int"
+        reveal_type(user.my_custom_field1)  # N: Revealed type is "myapp.models.CustomFieldValue"
+        reveal_type(user.my_custom_field2)  # N: Revealed type is "myapp.models.CustomFieldValue"
+        reveal_type(user.my_custom_field3)  # N: Revealed type is "builtins.bool"
+        reveal_type(user.my_custom_field4)  # N: Revealed type is "myapp.models.CustomFieldValue"
+        reveal_type(user.my_custom_field5)  # N: Revealed type is "Union[myapp.models.CustomFieldValue, None]"
+        # user.my_custom_field6 is incorrectly typed as non-optional
+        # reveal_type(user.my_custom_field6) ## N: Revealed type is "Union[myapp.models.CustomFieldValue, None]"
```

Comment:
This is incorrect on master too and does not pick up on null=True. When debugging, the instance seemed correct and the args were optional so I think this issue is coming from another piece of code somewhere.

Reply:
Slightly off topic but still related:

I think #1900 is involved and/or need to be considered regarding `null=True` and optional.

IMO we should finish explore that approach primarily, before moving on to a plugin approach.

Reply:
Would really like to see https://github.com/typeddjango/django-stubs/pull/1900 making it as well, although in its current form it doesn't really simplify creating custom field subclasses (and I believe addressing https://github.com/typeddjango/django-stubs/pull/1900#issuecomment-1878363899 would help in this case but requires significant work on mypy)